libpod/lock/lock.go

```diff
@@ -24,6 +24,9 @@ type Manager interface {
 	// The underlying lock MUST be the same as another other lock with the
 	// same UUID.
 	RetrieveLock(id uint32) (Locker, error)
+	// FreeAllLocks frees all allocated locks, in preparation for lock
+	// reallocation.
+	FreeAllLocks() error
```

Comment:
The disclaimer may be worth adding here as well with some notes why and when it's necessary to be called (mostly thinking about godoc.org).